### PR TITLE
Finished cleanup of System.Slices

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
The package now contains:
1. SpanExtensions
2. Memory<T> and related

At some point we should review SpanExtensions and move those we want to keep to System.Memory.dll

Memory<T> and related will be also moved to System.Memory.dll once the design stabilizes.

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
cc @AtsushiKan, @joshfree, @botaberg

File: src/System.Slices/System/SpanExtensions_binary.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public static partial class SpanExtensions
+    {
+        /// <summary>
+        /// Casts a Slice of one primitive type (T) to another primitive type (U).
+        /// These types may not contain managed objects, in order to preserve type
+        /// safety.  This is checked statically by a Roslyn analyzer.
+        /// </summary>
+        /// <param name="slice">The source slice, of type T.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Span<U> Cast<[Primitive]T, [Primitive]U>(this Span<T> slice)
```
Comment (jkotas) [inline src/System.Slices/System/SpanExtensions_binary.cs]: BTW: What makes the `[Primitive]` part compile today?

Comment (KrzysztofCwalina) [inline src/System.Slices/System/SpanExtensions_binary.cs]: The C# compiler :-), i.e. I am not sure what you are asking. It's stock C# syntax. It has no effect though.

File: src/System.Slices/System/SpanExtensions_array.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        /// </exception>
+        public static Span<T> Slice<T>(this T[] array, int start)
+        {
+            return new Span<T>(array, start);
+        }
+
+        /// <summary>
+        /// Creates a new slice over the portion of the target array beginning
+        /// at 'start' index and ending at 'end' index (exclusive).
+        /// </summary>
```
Comment (cH40z-Lord) [inline src/System.Slices/System/SpanExtensions_array.cs]: There's no `end` Index. It's calculated from `start` + `length`, no?

File: src/System.Slices/System/SpanExtensions_binary.cs (hunk 1 above)
Comment (jkotas) [inline src/System.Slices/System/SpanExtensions_binary.cs]: I see ... it just an attribute. I have not realized that you can stick attributes on generic parameters in C#.

Comment (KrzysztofCwalina) [inline src/System.Slices/System/SpanExtensions_binary.cs]: Yeah, I was thinking about removing this attribute during the cleanup, but I decided to wait till we design this C# feature. 

cc: @jaredpar

**Pull request merged**